right/src/secondary_role_driver.c

```diff
@@ -173,6 +178,12 @@ static secondary_role_state_t resolveCurrentKeyRoleIfDontKnowTimeout()
 
     //handle trigger by press
     if (Cfg.SecondaryRoles_AdvancedStrategyTriggerByPress) {
+        bool actionKeyWasPressedAndFromTheSameHalf = isPrimaryFromSameHalf && actionPress != NULL && Utils_AreKeysOnTheSameHalf(actionKeyId, dualRoleId);
+
+        if(actionKeyWasPressedAndFromTheSameHalf) {
```

Comment:
```suggestion
        if (actionKeyWasPressedAndFromTheSameHalf) {
```

An opinion-dependent nitpick, but laszlo is very keen on these.